common/djangoapps/third_party_auth/tests/test_provider.py

```diff
@@ -62,6 +62,17 @@ def test_enabled_returns_list_of_enabled_providers_sorted_by_name(self):
         with patch('third_party_auth.provider._PSA_OAUTH2_BACKENDS', backend_names):
             self.assertEqual(sorted(provider_names), [prov.name for prov in provider.Registry.enabled()])
 
+    def test_providers_available_for_login(self):
+        hidden_provider = self.configure_google_provider(visible=False, enabled=True)
+        normal_provider = self.configure_facebook_provider(visible=True, enabled=True)
+        implicitly_hidden_provider = self.configure_linkedin_provider(enabled=True)
+        disabled_provider = self.configure_twitter_provider(visible=True, enabled=False)
+        provider_ids = [provider.provider_id for provider in pipeline.accepting_logins()]
+        self.assertNotIn(hidden_provider.provider_id, provider_ids)
+        self.assertNotIn(implicitly_hidden_provider.provider_id, provider_ids)
+        self.assertNotIn(disabled_provider.provider_id, provider_ids)
+        self.assertIn(normal_provider.provider_id, provider_ids)
```

Comment:
This test is essentially testing `available_for_login`, right? Does `pipeline.accepting_logins()` only return `ProviderConfig` models that already have `self.accepts_logins` set to True? Just checking to see why `accepts_logins` isn't being set for the providers in this test.